bitsandbytes/functional.py

```diff
@@ -858,49 +861,39 @@ def quantize_blockwise(
 
     if absmax is None:
         n = A.numel()
-        blocks = n // blocksize
-        blocks += 1 if n % blocksize > 0 else 0
+        blocks = -(n // -blocksize)
         absmax = torch.zeros((blocks,), device=A.device, dtype=torch.float32)
 
     if out is None:
         out = torch.zeros_like(A, dtype=torch.uint8)
 
     if A.device.type != "cpu":
         assert blocksize in [4096, 2048, 1024, 512, 256, 128, 64]
-        cblocksize = ct.c_int32(blocksize)
-        prev_device = pre_call(A.device)
+
         code = code.to(A.device)
         is_on_gpu([code, A, out, absmax])
-        if A.dtype == torch.float32:
-            lib.cquantize_blockwise_fp32(
-                get_ptr(code),
-                get_ptr(A),
-                get_ptr(absmax),
-                get_ptr(out),
-                cblocksize,
-                ct.c_int(A.numel()),
-            )
-        elif A.dtype == torch.float16:
-            lib.cquantize_blockwise_fp16(
-                get_ptr(code),
-                get_ptr(A),
-                get_ptr(absmax),
-                get_ptr(out),
-                cblocksize,
-                ct.c_int(A.numel()),
-            )
-        elif A.dtype == torch.bfloat16:
-            lib.cquantize_blockwise_bf16(
+
+        fn_map = {
+            torch.float32: "cquantize_blockwise_fp32",
+            torch.bfloat16: "cquantize_blockwise_bf16",
+            torch.float16: "cquantize_blockwise_fp16",
+        }
+
+        if A.dtype not in fn_map.keys():
+            raise ValueError(f"Blockwise quantization only supports 16/32-bit floats, but got {A.dtype}")
+
+        fn = fn_map[A.dtype]
+
+        with torch.cuda.device_of(A):
+            lib[fn](
```

Comment:
```suggestion
        fn_map = {
            torch.float32: "cquantize_blockwise_fp32",
            torch.bfloat16: "cquantize_blockwise_bf16",
            torch.float16: "cquantize_blockwise_fp16",
        }

        fn = getattr(lib, fn_map.get(A.dtype), None)
        if not fn:
             raise ValueError(f"Blockwise quantization only supports 16/32-bit floats, but got {A.dtype}")

        with torch.cuda.device_of(A):
            fn(
```

to avoid the silly `lib[fn]`. `fn_map` should probably also be module-level so it isn't recomputed in this function over and over again.

Reply:
I went ahead and removed the lookup to keep this simple; maybe something we can do to further clean up in a separate PR.